File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/DanhMucTinTucForm.ts
import { BooleanEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface DanhMucTinTucForm {
    TenDm: StringEditor;
    Slug: StringEditor;
    MoTa: StringEditor;
    ThuTu: IntegerEditor;
    HienThi: BooleanEditor;
}

export class DanhMucTinTucForm extends PrefixedContext {
    static readonly formKey = 'SanPhamDB.DanhMucTinTuc';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DanhMucTinTucForm.init) {
            DanhMucTinTucForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = BooleanEditor;

            initFormType(DanhMucTinTucForm, [
                'TenDm', w0,
                'Slug', w0,
                'MoTa', w0,
                'ThuTu', w1,
                'HienThi', w2
            ]);
        }
    }
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/DanhMucTinTuc/DanhMucTinTucForm.cs
namespace wheystore.SanPhamDB.Forms;

[FormScript("SanPhamDB.DanhMucTinTuc")]
[BasedOnRow(typeof(DanhMucTinTucRow), CheckNames = true)]
public class DanhMucTinTucForm
{
    public string TenDm { get; set; }
    public string Slug { get; set; }
    public string MoTa { get; set; }
    public int ThuTu { get; set; }
    public bool HienThi { get; set; }
}